src/rust-crypto/rust-crypto.ts

```diff
@@ -1508,6 +1508,11 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, CryptoEventH
         // Disable backup, and delete all the backups from the server
         await this.backupManager.deleteAllKeyBackupVersions();
 
+        // Remove the stored secrets in the secret storage
+        await this.secretStorage.store("m.cross_signing.master", null);
+        await this.secretStorage.store("m.cross_signing.self_signing", null);
+        await this.secretStorage.store("m.cross_signing.user_signing", null);
+        await this.secretStorage.store("m.megolm_backup.v1", null);
         // Disable the recovery key and the secret storage
```

Comment:
To be complete we probably want to delete the default key too. E.g if `m.secret_storage.default_key` is `oWU8hN8nuCBJEc6WYaQSVDorQZ8syJet` we want to delete `m.secret_storage.key.oWU8hN8nuCBJEc6WYaQSVDorQZ8syJet`

Reply:
Where's the test 😜 !

Reply:
@BillCarsonFr  it was done here https://github.com/matrix-org/matrix-js-sdk/pull/4683/commits/310f2bbc57f4a6dbe2439c6ed4819422ce7022c3